tests/system/small/test_dataframe.py

```diff
@@ -3020,6 +3020,21 @@ def test_loc_select_with_column_condition(scalars_df_index, scalars_pandas_df_in
     )
 
 
+def test_loc_select_with_column_condition_bf_series(
+    scalars_df_index, scalars_pandas_df_index
+):
+    bf_result = scalars_df_index.loc[
+        :, series.Series(scalars_df_index.dtypes == "Int64")
```

Comment:
Could we do a test that derives from `scalars_df_index` instead? I'm a little worried about index alignment.

Reply:
Done. But it identifies another problem b/347072677. May need SQL to fix.